test/model.test.js

```diff
@@ -5356,6 +5356,88 @@ describe('Model', function() {
       return doc.save({ safe: { w: 0 } });
     });
 
+    it('save() with acknowledged writes fail if topology is not replica set (gh-6862)', function(done) {
+      // If w > 1 and there is no replica sets, mongodb will throw BadValue error
+      // This test uses this to check if option `w` is correctly propagated to mongodb
+
+      // skip this test if the server is a replica set
+      if (db.client.topology.constructor.name === 'ReplSet') {
+        return this.skip();
+      }
+
+      const schemaA = new Schema({
+        name: String
+      }, { writeConcern: { w: 2 }});
+      const schemaB = new Schema({
+        name: String
+      });
+
+      const UserA = db.model('gh6862_1', schemaA);
+      const UserB = db.model('gh6862_2', schemaB);
+
+      const userA = new UserA();
+      const userB = new UserB();
+      userA.save(function(error) {
+        assert.ok(error);
+        assert.equal(error.name, 'MongoError');
+        assert.equal(error.codeName, 'BadValue');
+
+        userB.save({ w: 2 },function(error) {
+          assert.ok(error);
+          assert.equal(error.name, 'MongoError');
+          assert.equal(error.codeName, 'BadValue');
+          done();
+        });
+      });
+    });
+
+    it.skip('save() with wtimeout defined in schema (gh-6862)', function(done) {
+      // If you want to test this, setup replica set with 1 primary up and 1 slave down
```

Comment:
The correct term is "secondary", not "slave". Minor detail but we should use the correct term.

Reply:
Yes we should definitely use the correct term. Please fix this in a future commit.